sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q12.sf100/explain.txt

```diff
@@ -142,15 +142,15 @@ BroadcastExchange (28)
 Output [2]: [d_date_sk#13, d_date#24]
 Batched: true
 Location [not included in comparison]/{warehouse_dir}/date_dim]
-PushedFilters: [IsNotNull(d_date), GreaterThanOrEqual(d_date,1999-02-22), LessThanOrEqual(d_date,1999-03-24), IsNotNull(d_date_sk)]
+PushedFilters: [IsNotNull(d_date), GreaterThanOrEqual(d_date,1999-02-22), IsNotNull(d_date_sk)]
```

Comment:
do we break filter pushdown in this PR?

Reply:
As I mentioned [here](https://github.com/apache/spark/pull/32949#discussion_r653588113) it seems so.
This is the most concerning point of this change.

Reply:
> cast('1999-02-22' AS DATE) + INTERVAL 30 days is TimestampType

I checked the corresponding code in `ResolveBinaryArithmetic`
```
case (DateType, DayTimeIntervalType(DAY, DAY)) => DateAdd(l, ExtractANSIIntervalDays(r))
```
`DateAdd` returns Date, I think we are fine now?

Reply:
Oh, return types have been changed. I'll re-run TPCDS again.